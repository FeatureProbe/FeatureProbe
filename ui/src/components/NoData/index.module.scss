.no-data {
  text-align: center;
  font-family: PingFangSC-Medium;
  font-size: 18px;
  color: #7d7d7d;
}

.no-data-image {
  width: 165px;
  margin: 150px 0 8px;
}