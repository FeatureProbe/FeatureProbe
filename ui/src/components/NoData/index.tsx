import { FormattedMessage } from 'react-intl';
import styles from './index.module.scss';

interface IProps {
  text?: string;
}

const NoData = (props: IProps) => {
  const { text } = props;

  return (
    <div className={styles['no-data']}>
      <div>
        <img className={styles['no-data-image']} src={require('images/no-data.png')} alt='no-data' />
      </div>
      <div>
        { text || <FormattedMessage id='common.nodata.text' />  }
      </div>
    </div>
  );
};

export default NoData;
