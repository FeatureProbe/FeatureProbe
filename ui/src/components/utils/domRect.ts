export class DOMRect {
    x: number;
    width: number;
    y: number;
    height: number;
    constructor(x: number, y: number, width: number, height: number) {
        this.x = x;
        this.y = y;
        this.width = width;
        this.height = height;
    }
}