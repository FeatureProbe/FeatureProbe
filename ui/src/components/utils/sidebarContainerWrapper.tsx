import { SidebarContainer } from 'layout/hooks';

export default SidebarContainer.Provider;
