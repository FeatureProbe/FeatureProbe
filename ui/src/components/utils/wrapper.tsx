import en_US from '../../locales/en-US.json';
import { IntlProvider } from 'react-intl';

export const IntlWrapper: React.FC = ({ children }) => {
  return (
    <IntlProvider locale="en" messages={en_US}>{children}</IntlProvider>
  );
};
