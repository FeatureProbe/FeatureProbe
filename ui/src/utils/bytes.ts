export function getBytes(str: string) {
  let bytes = str.length,
    i = 0;
  for (; i < bytes; i++) if (str.charCodeAt(i) > 255) bytes++;

  return bytes;
}

const MB = 1048576;
export const MAX_SIZE = MB;