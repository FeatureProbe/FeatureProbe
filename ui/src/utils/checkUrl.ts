function escapeStringRegexp(string: string): string {
  if (typeof string !== 'string') {
    throw new TypeError('Expected a string');
  }

  return string
    .replace(/[|\\{}()[\]^$+*?.]/g, '\\$&')
    .replace(/-/g, '\\x2d');
}

export function matchUrl(matcher: string, originUrl: string, targetUrl: string): boolean {
  let href = '';
  let hash = '';
  let search = '';

  try {
    const fullUrl = new URL(targetUrl, '');
    href = fullUrl.href;
    hash = fullUrl.hash;
    search = fullUrl.search;
  } catch (error) {
    href = targetUrl;
  }

  let regex;
  let testUrl;

  switch (matcher) {
    case 'EXACT':
      testUrl = href;
      regex = new RegExp('^' + escapeStringRegexp(originUrl) + '/?$');
      break;
    case 'SIMPLE':
      testUrl = href.replace(hash, '').replace(search, '');
      regex = new RegExp('^' + escapeStringRegexp(originUrl) + '/?$');
      break;
    case 'SUBSTRING':
      testUrl = href.replace(search, '');
      regex = new RegExp('.*' + escapeStringRegexp(originUrl) + '.*$');
      break;
    case 'REGULAR':
      testUrl = href.replace(search, '');
      regex = new RegExp(originUrl);
      break;
    default:
      return false;
  }

  return regex.test(testUrl);
}