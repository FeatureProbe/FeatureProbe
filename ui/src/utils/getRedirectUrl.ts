import { getProjectList } from 'services/project';
import { IProject, IEnvironment } from 'interfaces/project';
import { IDictionary } from 'interfaces/targeting';
import { getFromDictionary } from 'services/dictionary';
import { LAST_SEEN } from 'constants/dictionary_keys';

const getInfo = async () => {
  const res =  await getFromDictionary<IDictionary>(LAST_SEEN);
  const { success, data } = res;
  if (success && data) {
    return JSON.parse(data.value);
  } else {
    return {};
  }
};

export const getRedirectUrl = async (defaultRedirectUrl: string) => {
  let redirectUrl = defaultRedirectUrl;
  const info = await getInfo();
  const { projectKey, environmentKey } = info;

  if (!!projectKey && !!environmentKey) {
    redirectUrl = `/${projectKey}/${environmentKey}/toggles`;
  } else {
    const res = await getProjectList<IProject[]>();
    const { success, data } = res;
    if (success && data) {
      if (data[0]) {
        const { key, environments } = data[0];
        let envKey = '';
        environments.some((environment: IEnvironment) => {
          if (environment.key) {
            envKey = environment.key;
            return true;
          }
          return environment;
        });
  
        if (key && envKey) {
          redirectUrl = `/${key}/${envKey}/toggles`;
        }
      } else {
        redirectUrl = '/projects';
      }
    }
  }

  return redirectUrl;
};