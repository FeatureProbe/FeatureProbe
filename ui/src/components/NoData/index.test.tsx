import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import NoData from '.';
import { IntlWrapper } from 'components/utils/wrapper';

it('NoData snapshot', (done) => {
  (async () => {
    const { asFragment } = render(
      <NoData></NoData>
    , { wrapper: IntlWrapper});
    expect(asFragment()).toMatchSnapshot();
    done();
  })();
});

test('NoData custom text', (done) => {
  (async () => {
    render(
      <NoData text='No Data Test'></NoData>
    , { wrapper: IntlWrapper});

    screen.getByText('No Data Test');
    expect(screen.getByText('No Data Test')).not.toBeNull();
    done();
  })();
});
