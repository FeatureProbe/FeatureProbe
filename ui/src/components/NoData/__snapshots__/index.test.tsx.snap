// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoData snapshot 1`] = `
<DocumentFragment>
  <div
    class="no-data"
  >
    <div>
      <img
        alt="no-data"
        class="no-data-image"
        src="no-data.png"
      />
    </div>
    <div>
      No Data
    </div>
  </div>
</DocumentFragment>
`;
